test/common/http/http3_status_tracker_test.cc

```diff
@@ -0,0 +1,170 @@
+  tracker_.markHttp3Broken();
+
+  timer_->invokeCallback();
+  EXPECT_FALSE(tracker_.isHttp3Broken());
+  EXPECT_FALSE(tracker_.isHttp3Confirmed());
+
+  EXPECT_CALL(*timer_, enableTimer(std::chrono::milliseconds(20 * 60 * 1000), nullptr));
+  tracker_.markHttp3Broken();
+  EXPECT_TRUE(tracker_.isHttp3Broken());
+  EXPECT_FALSE(tracker_.isHttp3Confirmed());
+
+  timer_->invokeCallback();
+  EXPECT_FALSE(tracker_.isHttp3Broken());
+  EXPECT_FALSE(tracker_.isHttp3Confirmed());
+
+  EXPECT_CALL(*timer_, enableTimer(std::chrono::milliseconds(40 * 60 * 1000), nullptr));
+  tracker_.markHttp3Broken();
+  EXPECT_TRUE(tracker_.isHttp3Broken());
+  EXPECT_FALSE(tracker_.isHttp3Confirmed());
+
+  timer_->invokeCallback();
+  EXPECT_FALSE(tracker_.isHttp3Broken());
+  EXPECT_FALSE(tracker_.isHttp3Confirmed());
+}
+
+TEST_F(Http3StatusTrackerTest, MarkBrokenWithBackoffMax) {
+  // markBroken will only be called when the time is not enabled.
+  EXPECT_CALL(*timer_, enabled()).WillRepeatedly(Return(false));
+
+  EXPECT_CALL(*timer_, enableTimer(std::chrono::milliseconds(5 * 60 * 1000), nullptr));
+  tracker_.markHttp3Broken();
+  timer_->invokeCallback();
+
+  EXPECT_CALL(*timer_, enableTimer(std::chrono::milliseconds(10 * 60 * 1000), nullptr));
+  tracker_.markHttp3Broken();
+  timer_->invokeCallback();
+
+  EXPECT_CALL(*timer_, enableTimer(std::chrono::milliseconds(20 * 60 * 1000), nullptr));
+  tracker_.markHttp3Broken();
+  timer_->invokeCallback();
+
+  EXPECT_CALL(*timer_, enableTimer(std::chrono::milliseconds(40 * 60 * 1000), nullptr));
+  tracker_.markHttp3Broken();
+  timer_->invokeCallback();
+
+  EXPECT_CALL(*timer_, enableTimer(std::chrono::milliseconds(80 * 60 * 1000), nullptr));
+  tracker_.markHttp3Broken();
+  timer_->invokeCallback();
+
+  EXPECT_CALL(*timer_, enableTimer(std::chrono::milliseconds(160 * 60 * 1000), nullptr));
+  tracker_.markHttp3Broken();
+  timer_->invokeCallback();
+
+  EXPECT_CALL(*timer_, enableTimer(std::chrono::milliseconds(320 * 60 * 1000), nullptr));
+  tracker_.markHttp3Broken();
+  timer_->invokeCallback();
+
+  EXPECT_CALL(*timer_, enableTimer(std::chrono::milliseconds(640 * 60 * 1000), nullptr));
+  tracker_.markHttp3Broken();
+  timer_->invokeCallback();
+
+  EXPECT_CALL(*timer_, enableTimer(std::chrono::milliseconds(1280 * 60 * 1000), nullptr));
+  tracker_.markHttp3Broken();
+  timer_->invokeCallback();
+
+  // Broken period no longer increases.
+  EXPECT_CALL(*timer_, enableTimer(std::chrono::milliseconds(1280 * 60 * 1000), nullptr));
+  tracker_.markHttp3Broken();
+  timer_->invokeCallback();
+}
+
+TEST_F(Http3StatusTrackerTest, MarkBrokenThenExpiresThenConfirmedThenBroken) {
+  EXPECT_CALL(*timer_, enabled()).WillOnce(Return(false));
+  EXPECT_CALL(*timer_, enableTimer(std::chrono::milliseconds(5 * 60 * 1000), nullptr));
+  tracker_.markHttp3Broken();
+
+  timer_->invokeCallback();
+
+  EXPECT_CALL(*timer_, enabled()).WillOnce(Return(false));
```

Comment:
CI shows a test failure in this line.  I think this is due to enabled() being used in an ASSERT which is compiled out in release mode; mock expectations are not met when tests run in opt mode.